nixos/modules/services/x11/clight.nix

```diff
@@ -0,0 +1,112 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.clight;
+
+  toConf = { isAttr ? false }@args: v:
+    if builtins.isFloat v then toString v
+    else if isList v then "[ " + concatMapStringsSep ", " (toConf {}) v + " ]"
+    else if isAttr && isString v then toString v
+    else generators.toPretty {} v;
```

Comment:
Once the type is fully specified, there should be enough few possible types here that you probably don't need to use generators (which I don't like because its format isn't specified).